pkg/paravirt/paravirt_protected_entity_type_manager.go

```diff
@@ -166,3 +186,230 @@ func (this *ParaVirtProtectedEntityTypeManager) getDataTransports(id astrolabe.P
+			Resources:   resources,
+			VolumeMode:  volumeMode,
+			VolumeName:  pv.Name,
+		},
+	}
+
+	// Create a PVC in Guest Cluster to statically bind to PV in Guest Cluster
+	if pvc, err = this.gcKubeClientSet.CoreV1().PersistentVolumeClaims(namespace).Create(context.TODO(), pvc, metav1.CreateOptions{}); err == nil || apierrs.IsAlreadyExists(err) {
+		// Save succeeded.
+		if err != nil {
+			this.logger.Infof("PVC %s/%s already exists, reusing", pvc.Namespace, pvc.Name)
+			err = nil
+		} else {
+			this.logger.Infof("PVC %s/%s saved", pvc.Namespace, pvc.Name)
+		}
+	}
+	this.logger.Infof("CreateFromMeta: PVC %s/%s is created in the guest cluster", pvc.Namespace, pvc.Name)
+
+	// Wait for PVC and PV to bind to each other
+	err = astrolabe_pvc.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, this.gcKubeClientSet, pvc.Namespace, pvc.Name, astrolabe_pvc.Poll, astrolabe_pvc.ClaimBindingTimeout, this.logger)
+	if err != nil {
+		return nil, fmt.Errorf("pvc %s/%s did not become Bound: %v", pvc.Namespace, pvc.Name, err)
+	}
+
+	this.logger.Infof("CreateFromMetadata: PVC %s/%s is bound to PV %s.", pvc.Namespace, pvc.Name, pv.Name)
+
+	// Get CloneFromSnapshot from Guest
+	gcClone, err := this.gcBackupDriverClient.CloneFromSnapshots(cloneFromSnapshotNamespace).Get(context.TODO(), cloneFromSnapshotName, metav1.GetOptions{})
+	if err != nil {
+		this.logger.Errorf("Failed to get the cloneFromSnapshot CR %s/%s in Guest Cluster: %v", cloneFromSnapshotNamespace, cloneFromSnapshotName, err)
+		return nil, errors.Wrapf(err, "failed to get cloneFromSnapshot record in Guest Cluster")
+	}
+
+	// Update CloneFromSnapshot status in Guest based on CloneFromSnapshot from Supervisor
+	clone := gcClone.DeepCopy()
+	clone.Status.Phase = svcClone.Status.Phase
+	clone.Status.Message = svcClone.Status.Message
+	clone.Status.ResourceHandle = svcClone.Status.ResourceHandle.DeepCopy()
+	_, err = this.gcBackupDriverClient.CloneFromSnapshots(cloneFromSnapshotNamespace).UpdateStatus(context.TODO(), clone, metav1.UpdateOptions{})
+	if err != nil {
+		this.logger.Errorf("CreateFromMeta: Failed to update status of CloneFromSnapshot %s/%s to %v", cloneFromSnapshotNamespace, cloneFromSnapshotName, clone.Status.Phase)
+		return nil, err
+	}
+	this.logger.Infof("CreateFromMeta: CloneFromSnapshot %s/%s updated successfully to Phase %v", cloneFromSnapshotNamespace, cloneFromSnapshotName, clone.Status.Phase)
+
+	peID := astrolabe_pvc.NewProtectedEntityIDFromPVCName(pvc.Namespace, pvc.Name)
+	this.logger.Infof("CreateFromMetadata: generated peID: %s.", peID.String())
+
+	pe, err := this.GetProtectedEntity(ctx, peID)
+	if err != nil {
+		this.logger.WithError(err).Errorf("Failed to get the ProtectedEntity from peID %s", peID.String())
+		return nil, err
+	}
+
+	this.logger.Infof("CreateFromMetadata: retrieved ProtectedEntity for ID %s.", peID.String())
+	return pe, nil
+}
+
+func decodeSnapshotID(snapshotID astrolabe.ProtectedEntitySnapshotID, logger logrus.FieldLogger) (string, error) {
```

Comment:
There's a decodeSnapshotID in snapMgr, please use that, i don't think there's much of difference. Better still, move it to utils.